Add release notes for version 0.75.0, fix some typos (#354)
returntocorp/semgrep-docs: docs/release-notes.md

## November 2021

### Version 0.75.0

#### Fixes

##### Semgrep CI

> In Semgrep CI, the `--disable-nosem` now tags findings with the `is_ignored` option correctly. Previously, an optimization from version 0.74.0 left the field `None` when the described option has been used. The optimization has been reverted.

Reviewer: ```suggestion
In Semgrep CI, the option `--disable-nosem` now tags findings with the `is_ignored` option correctly. Previously, an optimization from version 0.74.0 left the field `None` when the described option has been used. The optimization has been reverted.
```

##### Semgrep CI no longer fails scan with binary files

> Before this update, under certain circumstances, Semgrep reported `Pcre.Error(BadUTF8) error` when it tried to analyze PNG, TTF, EOT or WOFF, zip, tar, and other binary files. As a consequence, scans failed when binary files were present. With this update, the underlying issue has been fixed, and Semgrep skips binary files. ([#4258](https://github.com/returntocorp/semgrep/issues/4258))

Reviewer: ```suggestion
Before this update, Semgrep sometimes reported `Pcre.Error(BadUTF8) error` when it tried to analyze PNG, TTF, EOT or WOFF, zip, tar, and other binary files. As a consequence, scans failed when binary files were present. With this update, the underlying issue has been fixed, and Semgrep skips binary files. ([#4258](https://github.com/returntocorp/semgrep/issues/4258))
```

## Finding remaining release notes

> This document encompasses only a limited number of release notes. See changelog that includes descriptions of older versions than displayed in this document: [Semgrep GitHub changelog](https://github.com/returntocorp/semgrep/releases)

Reviewer: ```suggestion
This document encompasses only a limited number of release notes. See the changelog that includes descriptions of older versions than displayed in this document: [Semgrep GitHub changelog](https://github.com/returntocorp/semgrep/releases)
```
